### ISSUE TITLE
kernel-lts -> linux-lts2019

### ISSUE THREAD

**Issue by Tornado-Tim** [clearlinux/distribution#1857]
Hi team,

Could a linux-lts2019-dev bundle be made or migrate linux-lts from the 2018 kernel to the 2019 kernel?

The 2018 LTS kernel is too old for my machine as it doesnt support my AMD GPU and I need to make and load dkms modules which does not work on the native kernel.

**Comment by ahkok** [comment]
Given the time frame (we're in march), we're at the point where we need to obsolete the 2017 lts kernel and bring in the 2019 lts kernel. That then would solve your issue here.

**Comment by mdhorn** [comment]
@miguelinux @ahkok The installer still offers the 2017 kernel, so please keep me in the loop if it is depreciated.

**Comment by bryteise** [comment]
@miguelinux We should consider only carrying a single lts kernel too.

**Comment by mdhorn** [comment]
> @miguelinux We should consider only carrying a single lts kernel too.

How "long" are we considering is "long term support"?
I'd think we'd need at least the previous two years?

**Comment by bryteise** [comment]
Well it isn't our long term support, just the kernel.org name (we'd just keep the newest "longterm" kernel in that case).

**Comment by fenrus75** [comment]
likely the installer should only offer (in gui) the latest lts


we offer the older ones as long as they are maintained, but we wouldn't
want new installs accidentally picking a kernel that's going away too
quickly
(and users can always post install switch)

**Comment by mdhorn** [comment]
> likely the installer should only offer (in gui) the latest lts we offer the older ones as long as they are maintained, but we wouldn't want new installs accidentally picking a kernel that's going away too quickly (and users can always post install switch)

The installer currently allows for selecting from the native kernel or any of the LTS options, but we could change that to just `kernel-native` and `kernel-lts` options.

**Comment by ahkok** [comment]
> we could change that to just kernel-native and kernel-lts options.

I would +1 that.

**Comment by bwarden** [comment]
The kernel-lts bundle just includes the kernel-lts2018 bundle. We already have a kernel-lts2019 bundle, so we should just change the include, right?

**Comment by mdhorn** [comment]
> The kernel-lts bundle just includes the kernel-lts2018 bundle. We already have a kernel-lts2019 bundle, so we should just change the include, right?

@miguelinux Was waiting on something before changing that definition -- not sure if it was just communications, or testing, or ???

**Comment by mdhorn** [comment]
> > we could change that to just kernel-native and kernel-lts options.
> 
> I would +1 that.

Installer change in the works
https://github.com/clearlinux/clr-installer/pull/693

**Comment by KyleSanderson** [comment]
@mdhorn NAK. Not sure what you guys are thinking... There were massive fuse bugs in LTS that were only just fixed in the last 2 weeks but still broken for months. https://github.com/trapexit/mergerfs/issues/686

Please resolve this terrible regression immediately, I purposely installed lts2018 for this reason.

**Comment by ahkok** [comment]
@KyleSanderson 

If there is a kernel patch we can merge on top of the 2019-lts, then we can (e.g. because it's in `fuse-next` already) just potentially do that. 

Also: You cannot "force" us to maintain a kernel beyond their life span. We will remove it. That was always the plan. Better work on a fix for the future now.

**Comment by bryteise** [comment]
@KyleSanderson we do want to ensure a good kernel version is being picked for the default. That said the 5.4 series has been around for about 4 months and so moving over default versions doesn't seem obviously problematic. While we would like to support kernels for as as long as they are supported upstream there is a cost to doing so. We aren't removing 4.19 immediately but stopping providing it as a target for the installer is the start of getting removed. I think do owe the community a timeline for the removal from the distro though.

**Comment by miguelinux** [comment]
The change is already done

**Comment by jonahbron** [comment]
I'm glad this update was done. However, the dkms bundle has not been updated. There needs to be a kernel-lts2019-dkms bundle added, and the kernel-lts-dkms bundle needs to point to it. 

Not having that has broken my ability to install the Nvidia driver.

**Comment by jonahbron** [comment]
Since there are other people upvoting my comment, it seems worth it to describe how I've worked around this. 

tl;dr the native kernel works fine, switch to it for now.

First run `sudo swupd bundle-install kernel-native`

Then run `sudo clr-boot-manager list-kernels`

Find the full name of the native kernel in that list. Use it to run `sudo clr-boot-manager set-kernel {name of kernel here}`

Reboot and go through the normal installation steps.

**Comment by bryteise** [comment]
A new kernel-lts2019-dkms bundle (and corresponding changes to kernel-lts-dkms) should be in the next day or so.

**Comment by jonahbron** [comment]
Confirmed that kernel-lts2019-dkms has been added.  If you (like me) switched to native for NVidia drivers, I recommend doing the following:

```bash
sudo swupd update
sudo swupd bundle-add kernel-lts kernel-lts-dkms
sudo clr-boot-manager set-kernel $(sudo clr-boot-manager list-kernels | grep lts2019 | tr -d ' ')
reboot
# Manually install NVidia drivers if desired
sudo swupd bundle-remove kernel-lts2018 kernel-lts2018-dkms
```
